fs2/src/main/scala/laserdisc/fs2/Frame.scala

```diff
@@ -0,0 +1,38 @@
+package laserdisc.fs2
+
+import cats.syntax.either._
+import fs2.Chunk
+import laserdisc.protocol.BitVectorDecoding._
+import laserdisc.protocol.RESP
+import laserdisc.|
+import scodec.bits.BitVector
+
+sealed trait Frame extends Product with Serializable {
+
+  def append(chunk: Chunk[Byte]): Exception | NonEmptyFrame =
+    nextFrame(BitVector.view(chunk.toByteBuffer))
+
+  protected final def nextFrame(bits: BitVector): Exception | NonEmptyFrame =
+    RESP.stateOf(bits) map {
+      case MissingBits(n)        => Incomplete(bits, n)
+      case CompleteAndDecoded(r) => Decoded(r.value)
+      case IncompleteVector      => Incomplete(bits, 0L)
+      case CompleteVector        => Complete(bits)
+    } leftMap (new Exception(_))
+}
+
+sealed trait NonEmptyFrame extends Product with Serializable
+case object Empty extends Frame
```

Comment:
`EmptyFrame` maybe?

Reply:
this is part of the Frame coproduct. For the others (`Complete`, `Decoded`, `Incomplete`) I haven't repeated the word *frame*. In case I would change all of them, what do you think.